src/main/java/seedu/letsgethired/model/InternTracker.java

```diff
@@ -111,7 +111,7 @@ public void setApplication(InternApplication target, InternApplication editedInt
      */
     public void removeApplication(InternApplication key) {
         internApplications.remove(key);
-        if (selectedApplication != null && !hasApplication(selectedApplication)) {
+        if (selectedApplication != null && selectedApplication.equals(key)) {
```

Comment:
Why was this change introduced ah?

Reply:
The condition always evaluated to false, because the application ended up being removed and then selectedApplication could not be set to null. This checks if the removed application is the one that was removed